homeassistant/components/device_tracker/meraki.py

```diff
@@ -0,0 +1,125 @@
+        """Initialize Meraki URL endpoints."""
+        self.async_see = async_see
+        self.validator = config[CONF_VALIDATOR]
+        self.secret = config[CONF_SECRET]
+
+    @asyncio.coroutine
+    def get(self, request):
+        """Meraki message received as GET."""
+        return self.validator
+
+    @asyncio.coroutine
+    def post(self, request):
+        """Meraki CMX message received."""
+        try:
+            data = yield from request.json()
+        except ValueError:
+            return self.json_message('Invalid JSON', HTTP_BAD_REQUEST)
+        _LOGGER.debug("Meraki Data from Post: %s", json.dumps(data))
+        if not data.get('secret', False):
+            _LOGGER.error("secret invalid")
+            return self.json_message('No secret', HTTP_UNPROCESSABLE_ENTITY)
+        if data['secret'] != self.secret:
+            _LOGGER.error("Invalid Secret received from Meraki")
+            return self.json_message('Invalid secret',
+                                     HTTP_UNPROCESSABLE_ENTITY)
+        elif data['version'] != VERSION:
+            _LOGGER.error("Invalid API version: %s", data['version'])
+            return self.json_message('Invalid version',
+                                     HTTP_UNPROCESSABLE_ENTITY)
+        else:
+            _LOGGER.debug('Valid Secret')
+            if data['type'] == "DevicesSeen":
+                _LOGGER.debug("WiFi Devices Seen")
+            elif data['type'] == "BluetoothDevicesSeen":
+                _LOGGER.debug("Bluetooth Devices Seen")
+            else:
+                _LOGGER.error("Unknown Device %s", type)
+                return self.json_message('invalid device type',
+                                         HTTP_UNPROCESSABLE_ENTITY)
+        if len(data["data"]["observations"]) == 0:
+            _LOGGER.debug("No observations found")
+            return
+        res = yield from self._handle(request.app['hass'], data)
+        return res
+
+    @asyncio.coroutine
+    def _handle(self, hass, data):
+        for i in data["data"]["observations"]:
+            data["data"]["secret"] = "hidden"
+            lat = i["location"]["lat"]
+            lng = i["location"]["lng"]
+            accuracy = int(float(i["location"]["unc"]))
+            mac = i["clientMac"]
+            _LOGGER.debug("clientMac: %s", mac)
+            gps_location = (lat, lng)
+            attrs = {}
+            if i.get('os', False):
+                attrs['os'] = i['os']
+            if i.get('manufacturer', False):
+                attrs['manufacturer'] = i['manufacturer']
+            if i.get('ipv4', False):
+                attrs['ipv4'] = i['ipv4']
+            if i.get('ipv6', False):
+                attrs['ipv6'] = i['ipv6']
+            if i.get('seenTime', False):
+                attrs['seenTime'] = i['seenTime']
+            if i.get('ssid', False):
+                attrs['ssid'] = i['ssid']
+            yield from self.async_see(
```

Comment:
make simple:
`hass.async_add_job(self.async_see(...))`

So it will not block on process and is faster and make that `_handle` will be a normal function they can add a decorator `@callback` from `core.py`.